packages/back-end/src/services/experiments.ts

```diff
@@ -806,6 +810,244 @@ export function toSnapshotApiInterface(
   };
 }
 
+/**
+ * While the `postMetricValidator` can detect the presence of values, it cannot figure out the correctness.
+ * @param payload
+ * @param datasource
+ */
+export function postMetricApiPayloadIsValid(
+  payload: z.infer<typeof postMetricValidator.bodySchema>,
+  datasource: Pick<DataSourceInterface, "type">
+): { valid: true } | { valid: false; error: string } {
+  const { type, sql, sqlBuilder, mixpanel, behavior } = payload;
+
+  // Validate query format: sql, sqlBuilder, mixpanel
+  let queryFormatCount = 0;
+  if (sqlBuilder) {
+    queryFormatCount++;
+  }
+  if (sql) {
+    queryFormatCount++;
+  }
+  if (mixpanel) {
+    queryFormatCount++;
+  }
+  if (queryFormatCount !== 1) {
+    return {
+      valid: false,
+      error: "Can only specify one of: sql, sqlBuilder, mixpanel",
+    };
+  }
+
+  // Validate behavior
+  if (behavior) {
+    if (
+      typeof behavior.riskThresholdDanger !== "undefined" &&
+      behavior.riskThresholdDanger < 0
+    )
+      return {
+        valid: false,
+        error: "riskThresholdDanger must be a non-negative number",
+      };
+
+    if (
+      typeof behavior.riskThresholdSuccess !== "undefined" &&
+      behavior.riskThresholdSuccess < 0
+    )
```

Comment:
should we check against `>1` as well? I can see people getting confused and entering "5" for 5% (which would actually be 500%)

Reply:
I don't think this is correct—we can have float values, and the docs say:

> Threshold for Risk to be considered low enough, as a proportion (e.g. put 0.0025 for 0.25%). <br/> Must be a non-negative number and must not be higher than `riskThresholdDanger`